sdk/metric/reader_test.go

```diff
@@ -106,11 +117,34 @@ func (ts *readerTestSuite) TestMultipleRegister() {
 	ts.Equal(assert.AnError, err)
 }
 
+func (ts *readerTestSuite) TestExternalProducerPartialSuccess() {
```

Comment:
should we add a test case for partial success that "sdk producer faild but external producer succeeded"?

Reply:
done